### PR TITLE
[core] Use longs instead of ints to support larger key spaces.

### PR DESCRIPTION
**Pull request by mairbek**

### CODE REVIEW

File: core/src/main/java/com/yahoo/ycsb/generator/UniformGenerator.java [hunk 1]
```diff
@@ -48,7 +48,8 @@ public UniformGenerator(Collection<String> values)
 	@Override
   public String nextValue()
 	{
-		_laststring = _values.get(_gen.nextValue());
+	  int val = _gen.nextValue().intValue();
```
Comment (risdenk) [inline core/src/main/java/com/yahoo/ycsb/generator/UniformGenerator.java]: Is this line necessary?

**Comment by risdenk** [comment]
@mairbek thanks for the PR. Just for reference this looks related to #664 and may supercede PR #126

File: core/src/main/java/com/yahoo/ycsb/generator/UniformGenerator.java [hunk 2]
```diff
@@ -48,7 +48,7 @@ public UniformGenerator(Collection<String> values)
 	@Override
   public String nextValue()
 	{
-		_laststring = _values.get(_gen.nextValue());
+		_laststring = _values.get(_gen.nextValue().intValue());
```
Comment (risdenk) [inline core/src/main/java/com/yahoo/ycsb/generator/UniformGenerator.java]: Is there a reason we are doing .intValue()? Sorry I wasn't clear earlier with my comment. Should we be using longs everywhere?

File: core/src/main/java/com/yahoo/ycsb/generator/AcknowledgedCounterGenerator.java [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2015 YCSB contributors. All rights reserved.
+ * Copyright (c) 2017 YCSB contributors. All rights reserved.
```
Comment (risdenk) [inline core/src/main/java/com/yahoo/ycsb/generator/AcknowledgedCounterGenerator.java]: These should be 2015-2017 and not a replacement. I should have been more clear.

File: core/src/main/java/com/yahoo/ycsb/generator/UniformGenerator.java (hunk 2 above)
Comment (mairbek) [inline core/src/main/java/com/yahoo/ycsb/generator/UniformGenerator.java]: The size of ArrayList is int, so limiting the value.

Also UniformGenerator doesn't seem to be used so maybe we can consider removing.

**Comment by mairbek** [comment]
Thanks for the review!

Thanks for pointing me to #126, it seems to be very similar.

I will update measurements related code too. Looking at OneMeasurementHistogram.java:60, number of operations is an Integer so that code could potentially go wrong. At this point we are not running more then 1B operations so I never saw it breaking before.

File: core/src/test/java/com/yahoo/ycsb/generator/AcknowledgedCounterGeneratorTest.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2015 YCSB contributors. All rights reserved.
+ * Copyright (c) 2016-2017 YCSB contributors. All rights reserved.
```
Comment (risdenk) [inline core/src/test/java/com/yahoo/ycsb/generator/AcknowledgedCounterGeneratorTest.java]: should be 2015-2017

File: core/src/main/java/com/yahoo/ycsb/workloads/RestWorkload.java [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2016 YCSB contributors. All rights reserved.
+ * Copyright (c) 2017 YCSB contributors. All rights reserved.
```
Comment (risdenk) [inline core/src/main/java/com/yahoo/ycsb/workloads/RestWorkload.java]: should be 2016-2017

**Comment by mairbek** [comment]
Updated the headers. Squashed all header related commits as well...

**Comment by risdenk** [comment]
This change looks good to me. @busbey or @cmccoy do you guys have any comments? I know you reviewed #126 and #432. This is just for the larger key spaces and doesn't touch operations.

**Comment by risdenk** [comment]
@mairbek - Sorry this looks like it conflicted with the checkstyle changes for core #895. Can you update your PR?

**Comment by mairbek** [comment]
Merged formatting changes.

**Comment by risdenk** [comment]
@mairbek sorry for the delay in this. I'm taking a look.

**Comment by mairbek** [comment]
@risdenk ping, it would be nice to get this in to be able to run larger scale benchmarks

**Comment by manolama** [comment]
Looked good, cleaned up the merge issues and committed in 59bc986a088a508a07144b425a9389ff6fc85843.

**Pull request closed without merging**